pkg/util/shard.go

```diff
@@ -24,7 +24,7 @@ var (
 // ShuffleShardSeed returns seed for random number generator, computed from provided identifier.
 func ShuffleShardSeed(identifier, zone string) int64 {
 	// Use the identifier to compute an hash we'll use to seed the random.
-	hasher := md5.New()
+	hasher := md5.New()               //nolint:gosec
```

Comment:
The seed is consumed in a way that it's expected to be equal for the same tenant across different query frontends:
```
// seed for shuffle sharding of queriers; computed from tenantID only,
// and is therefore consistent between different frontends.
```
Changing it means this assumption is broken during the rollout, which could have unintended side effects. Otherwise it's only used internally to seed in a way that's stable based on tenant ID, we do not rely on specifically a cryptographic hash here.